companion/src/eeprominterface.cpp

```diff
@@ -1357,21 +1357,74 @@ QList<FirmwareInterface *> firmwares;
 FirmwareInterface * default_firmware_variant;
 FirmwareInterface * current_firmware_variant;
 
-void UnregisterFirmwares() 
+void UnregisterFirmwares()
 {
   foreach (FirmwareInterface * f, firmwares) {
     delete f;
   }
 }
 
-bool LoadEeprom(RadioData &radioData, const uint8_t *eeprom, const int size)
+void ShowEepromErrors(QWidget *parent, QString title, QString mainMessage, unsigned long errorsFound)
+{
+  std::bitset<NUM_ERRORS> errors(errorsFound);
+  QString errorsList(QT_TRANSLATE_NOOP("EepromInterface", "Possible causes for this:\n"));
+
+  if (errors.test(UNSUPPORTED_NEWER_VERSION)) { errorsList += QT_TRANSLATE_NOOP("EepromInterface", "\n- Eeprom is from a newer version of OpenTX"); }
```

Comment:
Instead of the \n in all strings, I would see a QStringList with join("\n"), no?